app/js/lib/speech-controller.js

```diff
@@ -0,0 +1,98 @@
+import EventDispatcher from './common/event-dispatcher';
+import WakeWordRecognizer from './wakeword/recogniser.js';
+
+const p = Object.freeze({
+  wakewordRecognizer: Symbol('wakewordRecognizer'),
+  wakewordModelUrl:   Symbol('wakewordModelUrl'),
+});
+
+const EVENT_INTERFACE = [
+  // Emit when the wakeword is being listened for
+  'wakelistenstart',
+
+  // Emit when the wakeword is no longer being listened for
+  'wakelistenstop',
+
+  // Emit when the wakeword is heard
+  'wakeheard',
+
+  // Emit when the speech recognition engine starts listening
+  // (And _could_ be sending speech over the network)
+  'speechrecognitionstart',
+
+  // Emit when the speech recognition engine returns a recognised phrase
+  'speechrecognitionstop',
+];
+
+export default class SpeechController extends EventDispatcher {
+
+  constructor() {
+    super(EVENT_INTERFACE);
+
+    const recognizer = new WakeWordRecognizer();
+
+    recognizer.setOnKeywordSpottedCallback(
+                this._handleKeywordSpotted.bind(this));
```

Comment:
nit: Let's keep the alignment simple and only indent such lines with 2 spaces.
Let's fix it everywhere else too.